File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.py
# Last updated: 7/30/2025, 12:24:50 PM
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        mi = prices[0]
        profit = 0
        for i in prices :
            mi = min(i,mi)
            profit = max(profit,i-mi)
        return profit
        

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.java
// Last updated: 7/30/2025, 12:24:40 PM
class Solution {
    public int maxProfit(int[] prices) {
        int mi = prices[0];
        int profit=0;
        for(int p : prices){
            mi = Math.min(mi,p);
            profit = Math.max(profit,p-mi);
        }
        return profit;
        // for(int i = 1; i<prices.length; i++){
            
        // }
    }
}